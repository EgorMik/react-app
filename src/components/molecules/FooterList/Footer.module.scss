$color: rgb(58, 59, 59);
$grayColor: #333;

.footer {
    display:grid;
    width: 594px;
    grid-template-columns: 76px 168px 124px;
    align-items: start;
    justify-content: space-between;
    padding-top: 15px;
    padding-bottom: 15px;
    padding-left: 5px;
    z-index: 100;
    span {
        display: inline-block;
        padding: 8px 10px;
        margin: 0 5px;
        border-radius: 3px;
        color: $color;
    }
    a {
        color: $grayColor;
        text-decoration: none;
        margin: 0 5px;
    }
    p {
        float:left;
        margin: 0;
        color: rgb(145, 145, 145);
    }
}

.active{
    color:rgb(7, 19, 146);
    cursor: pointer;
    border: none;
    background: none;
    font-size:16px;
    font-family: 'Josefin Sans', sans-serif;
    &:hover { color:rgb(37, 37, 37);
        cursor: pointer;}
    }
.unactive{
    color: rgb(105, 105, 105);
    border: none;
    background: none;
    cursor: pointer;
    font-family: 'Josefin Sans', sans-serif;
    &:hover { color:rgb(37, 37, 37);
        cursor: pointer;}
    }