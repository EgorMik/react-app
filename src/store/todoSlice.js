import { createSlice} from '@reduxjs/toolkit';
export const filters = {
    ALL: "ALL",
    COMPLETED: "COMPLETED",
    NOT_COMPLETED: "NOT_COMPLETED"
  }
  const localStorageTodos = JSON.parse(localStorage.getItem('todo'));

if (!localStorageTodos) {
  const initalTodos = {
    todos: [ {
        id: '714ab407-82d7-4d96-ba8f-34c0f047acce',
        text: 'Complete online JavaScript course',
        completed: true,
      }],
  };

  localStorage.setItem('todo', JSON.stringify(initalTodos));
}

    
const todoSlice = createSlice({
    name:'todos',
    initialState:  JSON.parse(localStorage.getItem('todo')) !== null
    ? JSON.parse(localStorage.getItem('todo'))
    : { todos: [] },
    reducers:{
        addTodo(state, action){
            state.todos.push(action.payload)
            localStorage.setItem('todo', JSON.stringify(state));
       },
        removeTodo(state, action){
            state.todos = state.todos.filter(todo => todo.id !== action.payload.id);
           
            localStorage.setItem('todo', JSON.stringify(state));
        },
        toggleComplete(state, action){
            const toggledTodo = state.todos.find(todo => todo.id === action.payload.id);
            toggledTodo.completed = !toggledTodo.completed;
            localStorage.setItem('todo', JSON.stringify(state));
        },
        filterBy(state, action){
            state.filterBy = action.payload;
        },
        clearComplited(state, action){
            state.todos = state.todos.filter(todo => todo.completed !== action.payload);
            localStorage.setItem('todo', JSON.stringify(state));
        },
        dragAndDropSave(state, action) {
            state.todos = action.payload;
            localStorage.setItem('todo', JSON.stringify(state));
          },
    }
});
export const {addTodo,removeTodo,toggleComplete,filterBy,clearComplited,dragAndDropSave} = todoSlice.actions;
export default todoSlice.reducer;
