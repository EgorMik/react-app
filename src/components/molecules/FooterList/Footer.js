import React from 'react';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { filterBy, filters, clearComplited } from '../../../store/todoSlice';
import styles from './Footer.module.scss';
const Footer= ()=> {
    const dispatch = useDispatch();
    const todos = useSelector(state => state.todos);
    const filter = useSelector((state) => state.todos.filterBy);
    return (
        <div className={styles.footer}>
          <div>
            <p>{Object.values(todos.todos).filter(e=>e.completed === false).length} items left</p>
            </div>
          <div>
        <button onClick={() => dispatch(filterBy(filters.ALL))} className={filter=== "ALL" ? styles.active : styles.unactive }>All</button>
        <button onClick={() => dispatch(filterBy(filters.COMPLETED))} className={filter=== "COMPLETED" ? styles.active : styles.unactive }>Completed</button>
        <button onClick={() => dispatch(filterBy(filters.NOT_COMPLETED))} className={filter=== "NOT_COMPLETED" ? styles.active : styles.unactive }>
          Active
        </button>
        </div>
        <div>
        <button 
        className={styles.unactive}
        onClick={() => dispatch(clearComplited(true))} >
          Clear Compleated
        </button>
        </div>
      </div>
    );
  }
 
  export default Footer;