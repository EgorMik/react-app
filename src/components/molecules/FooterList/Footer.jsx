import React from 'react';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { filterBy, filters, clearComplited } from '../../../store/todoSlice';

const Footer= ({displayStyle })=> {
    const dispatch = useDispatch();
    const todos = useSelector(state => state.todos);
    return (
        <div className="footer">
          <div className="left">
            <p>{Object.keys(todos.todos.filter(e=>e.completed === false)).length} items left</p>
            </div>
            <div className="middle">
        <button onClick={() => dispatch(filterBy(filters.ALL))}>All</button>
        <button onClick={() => dispatch(filterBy(filters.COMPLETED))} >Completed</button>
        <button onClick={() => dispatch(filterBy(filters.NOT_COMPLETED))} >
          Active
        </button>
        </div>
        <div className="right">
        <button onClick={() => dispatch(clearComplited(true))} >
          Clear Compleated
        </button>
        </div>
      </div>
    );
  }
  export default Footer;