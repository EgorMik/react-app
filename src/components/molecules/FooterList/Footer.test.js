import React from "react";
import renderer from "react-test-renderer";
import FooterList from "./Footer";
import { Provider } from "react-redux";
import store from '../../../store'

describe("TodoList", () => {
    it("should renders correctly", () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <FooterList />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    })});
