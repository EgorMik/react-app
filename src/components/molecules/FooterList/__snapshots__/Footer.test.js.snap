// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should renders correctly 1`] = `
<div
  className="footer"
>
  <div>
    <p>
      0
       items left
    </p>
  </div>
  <div>
    <button
      className="unactive"
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="unactive"
      onClick={[Function]}
    >
      Completed
    </button>
    <button
      className="unactive"
      onClick={[Function]}
    >
      Active
    </button>
  </div>
  <div>
    <button
      className="unactive"
      onClick={[Function]}
    >
      Clear Compleated
    </button>
  </div>
</div>
`;
